os/arch/arm/src/armv7-m/up_blocktask.c

```diff
@@ -158,6 +162,15 @@ void up_block_task(struct tcb_s *tcb, tstate_t task_state)
 			save_task_scheduling_status(rtcb);
 #endif
 
+			/* Restore the MPU registers in case we are switching to an application task */
+#if defined(CONFIG_ARMV7M_MPU) && defined(CONFIG_APP_BINARY_SEPARATION)
+			if ((rtcb->flags & TCB_FLAG_TTYPE_MASK) != TCB_FLAG_TTYPE_KERNEL) {
+				putreg32(rtcb->mpu_regs[REG_RNR], MPU_RNR);
+				putreg32(rtcb->mpu_regs[REG_RBAR], MPU_RBAR);
+				putreg32(rtcb->mpu_regs[REG_RASR], MPU_RASR);
+			}
```

Comment:
How about moving this to a separate inline function or macro? Used at multiple places.

Reply:
OK. I created an inline function in mpu.h